spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -562,6 +614,102 @@ def set_unit_properties_by_max_channel_properties(recording, sorting, property,
                                                  ms_before=ms_before, ms_after=ms_after, verbose=verbose)
             sorting.set_unit_property(unit_id, property, recording.get_channel_property(max_chan, property))
 
+def get_firing_times_ids(sorting, sampling_frequency):
+    '''
+    Computes and returns the spike times in seconds and also returns 
+    along with cluster_ids needed for quality metrics
+    
+    Parameters
+    ----------
+    sorting: SortingExtractor
+        The sorting extractor
+    sampling_frequency: float
+        The sampling frequency of the recording
+
+    Returns
+    -------
+    spike_times: numpy.ndarray (num_spikes x 0)
+        Spike times in frames
+    spike_clusters: numpy.ndarray (num_spikes x 0)
+        Cluster IDs for each spike time
+    channel_map: numpy.ndarray (num_units x 0)
+        Original data channel for pc_feature_ind array
+    '''
+
+    # spike times.npy and spike clusters.npy
+    spike_times = np.array([])
+    spike_clusters = np.array([])
+
+    for i_u, id in enumerate(sorting.get_unit_ids()):
+        st = sorting.get_unit_spike_train(id)
+        cl = [i_u] * len(sorting.get_unit_spike_train(id))
+        spike_times = np.concatenate((spike_times, np.array(st)))
+        spike_clusters = np.concatenate((spike_clusters, np.array(cl)))
+
+    sorting_idxs = np.argsort(spike_times)
+    spike_times = spike_times[sorting_idxs, np.newaxis]
+    spike_clusters = spike_clusters[sorting_idxs, np.newaxis]
+
+    spike_times = (spike_times/sampling_frequency).flatten('F')
+    spike_clusters = spike_clusters.astype(int).flatten('F')
+
+    return spike_times, spike_clusters
+
+def get_quality_metric_data(recording, sorting, nPC=3, ms_before=1., ms_after=2., dtype=None, 
```

Comment:
I think that this would belong to a valitation_tools. In fact I don't see how this function would be called outside the validation module. Instead the other postprocessing functions are more general/standalone

Reply:
Agree